src/client/transaction.go

```diff
@@ -35,6 +35,30 @@ func (c APIClient) WithTransaction(txn *transaction.Transaction) *APIClient {
 	return c.WithCtx(ctx)
 }
 
+// WithoutTransaction returns a new APIClient which will run all future operations
+// outside of any active transaction
+// Removing from both incoming and ougoing metadata is necessary because Ctx() merges them
```

Comment:
This feels really messy, but I don't see a way around it as long as Ctx() indiscriminately combines metadata. I could also explicitly skip that key in AddMetadata but I wasn't confident about all GRPC paths that we care about.

Reply:
I don't like that we have to do this, but this is ~fine for now in my opinion.